api/runnerpool/ch_placer.go

```diff
@@ -49,11 +49,17 @@ func (p *chPlacer) PlaceCall(ctx context.Context, rp RunnerPool, call RunnerCall
 	var runnerPoolErr error
 	for {
 		var runners []Runner
-		runners, runnerPoolErr = rp.Runners(ctx, call)
-
-		i := int(jumpConsistentHash(sum64, int32(len(runners))))
-		for j := 0; j < len(runners) && !state.IsDone(); j++ {
+		for j := 0; !state.IsDone(); j++ {
+			// refresh each iteration (should be cached...). detect if runner list changed, and reset j if so (?)
+			oldRunners := runners
+			runners, runnerPoolErr = rp.Runners(ctx, call)
+			if !equal(runners, oldRunners) {
+				j = 0
+			}
+			i := int(jumpConsistentHash(sum64, int32(len(runners))))
+			i = (i + j) % len(runners)
```

Comment:
@jan-g @JadeRedworth i think this will aid the test results quite a bit, I could see why things were egregious... previously we did this:

```go
runners, runnerPoolErr = rp.Runners(ctx, call)
i := int(jumpConsistentHash(sum64, int32(len(runners))))
for j := 0; j < len(runners) && !state.IsDone(); j++ {
  r := runners[i]
  placed, err := state.TryRunner(r, call)
  // ...
}
```

meaning we were trying the same `runner=runners[i]` for `n=len(runners)` times, ouchy